sympy/core/power.py

```diff
@@ -407,7 +407,14 @@ def _eval_Mod(self, q):
             exponentiation.
             '''
             if self.base.is_Integer and self.exp.is_Integer and q.is_Integer:
-                return pow(int(self.base), int(self.exp), int(q))
+                b, e, m = int(self.base), int(self.exp), int(q)
+                # for very large powers
+                if m<=10**18 and _log(e,2)>4*m**(1/4): # pollard rho for modulo would not be slower than binary exponentiation
```

Comment:
What is the connection with Pollard rho?

Reply:
totient function would be calling factorint for evaluation, so, pollard rho.